design-documents/graph-ql/coverage/gift-registry.graphqls

```diff
@@ -0,0 +1,324 @@
+}
+
+type GiftRegistryRoleAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface, GiftRegistrySearcheableAttributeMetadataInterface, GiftRegistrySelectAttributeMetadataInterface {
+}
+
+interface GiftRegistrySearcheableAttributeMetadataInterface {
+    is_searcheable: Boolean!
+}
+
+interface GiftRegistrySelectAttributeMetadataInterface {
+    options: [GiftRegistrySelectAttributeOptionMetadata]
+}
+
+type GiftRegistrySelectAttributeOptionMetadata {
+    code: ID!
+    label: String!
+    is_default: Boolean
+}
+
+type GiftRegistry {
+    id: ID!
+    event_name: String!
+    type: GiftRegistryType
+    message: String!
+    created_on: String! @doc(description: "Creation date") @doc(description: "Accessible to the registry owner only")
+    privacy_settings: GiftRegistryPrivacySettings! @doc(description: "Accessible to the registry owner only")
+    status: GiftRegistryStatus! @doc(description: "Accessible to the registry owner only")
+    owner_name: String!
+    registrants: [GiftRegistryRegistrant]
+    shipping_address: CustomerAddress @doc(description: "Accessible to the registry owner only")
+    dynamic_attributes: [GiftRegistryDynamicAttribute]
+    items: [GiftRegistryItemInterface]
+}
+
+interface GiftRegistryItemInterface {
+    id: String!
+    quantity: Float!
+    quantity_fulfilled: Float!
+    note: String
+    added_on: String!
+    product: ProductInterface
+    customizable_options: [SelectedCustomizableOption]
+}
+
+type SimpleGiftRegistryItem implements GiftRegistryItemInterface {
+}
+
+type BundleGiftRegistryItem implements GiftRegistryItemInterface {
+    bundle_options: [SelectedBundleOption!]
+}
+
+type ConfigurableGiftRegistryItem implements GiftRegistryItemInterface {
+    child_sku: String! @doc(description: "SKU of the simple product corresponding to a set of selected configurable options.")
+    configurable_options: [SelectedConfigurableOption!]
+}
+
+# Not currently supported by Magento core
+type DownloadableGiftRegistryItem implements GiftRegistryItemInterface {
+    links: [DownloadableProductLink]
+    samples: [DownloadableProductSamples]
+}
+
+# Not currently supported by Magento core
+type VirtualGiftRegistryItem implements GiftRegistryItemInterface {
+}
+
+# Not currently supported by Magento core
+type GiftCardGiftRegistryItem implements GiftRegistryItemInterface {
+    sender_name: String!
+    recepient_name: String!
+    amount: SelectedGiftCardAmount
+    message: String
+}
+
+interface GiftRegistryDynamicAttributeInterface  {
+    code: ID!
+    label: String!
+    value: String!
+}
+
+type GiftRegistryDynamicAttribute implements GiftRegistryDynamicAttributeInterface {
+    group: GiftRegistryDynamicAttributeGroup!
+}
+
+type GiftRegistryRegistrantDynamicAttribute implements GiftRegistryDynamicAttributeInterface {
+
+}
+
+enum GiftRegistryDynamicAttributeGroup {
+    GENERAL_INFORMATION
+    EVENT_INFORMATION
+    PRIVACY_SETTINGS
+    DETAILED_INFORMATION
+    SHIPPING_ADDRESS
```

Comment:
Hi @paliarush one more question regarding the Attribute Groups. Currently we have the:
- Event Information
- Gift Registry Details
- Privacy Settings
- Shipping Address

So, I'm not sure what's the correct mapping here:
- Event Information -> `EVENT_INFORMATION`
- Gift Registry Details -> `DETAILED_INFORMATION` or `GENERAL_INFORMATION`
- Privacy Settings -> `PRIVACY_SETTINGS`
- Shipping Address -> `SHIPPING_ADDRESS`

Reply:
@eduard13 thanks for the question, please check the screenshot here https://docs.magento.com/user-guide/marketing/gift-registries.html
Try to configure all possible attributes with different attribute groups and see how they are displayed on the storefront, it is possible that current UX is not optimal and need to be improved, feel free to suggest how to improve it.